logger.js

```diff
@@ -141,18 +141,19 @@ logger.log = (level, data, logobj) => {
 
     // buffer until plugins are loaded
     const emptyPluginList = !plugins || Array.isArray(plugins.plugin_list) && !plugins.plugin_list.length;
+
     if (emptyPluginList) {
         logger.deferred_logs.push(item);
-        return true;
```

Comment:
One of the options P42 often suggests is early returns. Here we have one, and SonarLint (or you) have decided against the early return. I tend to favor early returns as I think it makes code more linear and easier to read. Here, the change strikes me as completely arbitrary. Either revert, or make the case that this change makes the code _better_ in some way.

Reply:
As I recall from this, the issue here was that all return paths were true, so this clarifies that fact.